cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -362,9 +363,10 @@ findProjectRoot = do
 readProjectConfig :: Verbosity -> FilePath -> Rebuild ProjectConfig
 readProjectConfig verbosity projectRootDir = do
     global <- readGlobalConfig verbosity
-    local  <- readProjectLocalConfig      verbosity projectRootDir
-    extra  <- readProjectLocalExtraConfig verbosity projectRootDir
-    return (global <> local <> extra)
+    local  <- readProjectLocalConfig       verbosity projectRootDir
+    freeze <- readProjectLocalFreezeConfig verbosity projectRootDir
```

Comment:
@23Skidoo so looking at the long discussion in that ticket, I'm not sure the objections there apply. It doesn't seem from that ticket that @tibbe really objects to having a separate file for freeze, but he's more concerned with UI complexity. Issues like, how do users know what files are involved, which ones do they edit etc. I think we can answer those questions quite clearly by using an explicit scheme like the above, where you can see how the top level project file includes others. Also, having comments in the relevant files will help make it clear.

It's true we have to make sure we're clear about the freeze workflow, but for the moment I think it's pretty much the same as before. We may want to add an unfreeze that just rm's the cabal.project.freeze file. I also don't foresee any problems with checking in the freeze file into source control.